addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2836,12 +2836,13 @@ const SnippetOptionWidget = Widget.extend({
                 return;
             }
 
-            await new Promise(resolve => setTimeout(() => {
+            await new Promise(resolve => setTimeout(async () => {
                 // Will update the UI of the correct widgets for all options
                 // related to the same $target/editor
-                this.trigger_up('snippet_option_update', {
-                    onSuccess: () => resolve(),
-                });
+                const pendingUpdates = [];
+                this.trigger_up('snippet_option_update', { pendingUpdates });
+                await Promise.all(pendingUpdates);
+                resolve();
```

Comment:
Stable changes = No API change :man_dancing: :woman_dancing: :man_dancing: :woman_dancing:

Anyways, even without compatibility change in mind, it could be prettier: for me the API should not change for the initial caller: I trigger an update notification and expect a call to `onSuccess` when it has been answered to. The initial caller should not have to care about building a pendingUpdates array and then waiting for all of them to be resolved.

Probably the simplest: the parent stops the propagation of the event and re-triggers it himself with its own created `onSuccess` method so that he knows when its own parent has finished dealing with the update. I would probably go that way, to test.